documentation/src/main/asciidoc/user_guide/chapter-63-Streams.adoc

```diff
@@ -0,0 +1,416 @@
+with less required data) or keeping track of wha keys it is has processed.
+
+It is possible but highly discouraged to disable rehash awareness on the stream.  This should only be considered if
+your request can handle only seeing a subset of data if a rehash occurs.  This can be done by invoking the
+link:https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/CacheStream.html#disableRehashAware--[disableRehashAware]
+method on the `CacheStream`.  The performance gain for most operations when a rehash doesn't occur is completely
+negigible.  The only exceptions are for iterator and forEach which will use less memory since they do not have
+to keep track of processed keys.
+
+WARNING: Please rethink disabling rehash awareness unless you really know what you are doing.
+
+==== Serialization / Externalization
+
+Since the operations are sent across to other nodes they must be externalizable by Infinispan marshalling.  This allows the
+operations to be sent to the other nodes.
+
+The simplest way is to cast your lambdas as Serializable as below.
+
+Unfortunately the link:https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html[Collectors]
+class doesn't produce Serializable instances.  Thus if you need to use these, you can use the newly provided
+link:http://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/stream/CacheCollectors.html[CacheCollectors]
+class which allows for a `Supplier<Collector>` to be provided.  This instance could then use the
+link:https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html[Collectors]
+to supply a `Collector`
+
+[source,java]
+----
+Map<Object, String> jbossValues = cache.entrySet().stream()
+              .filter((Serializable & Predicate<Map.Entry<Object, String>>) e -> e.getValue().contains("Jboss"))
+              .collect(CacheCollectors.serializableCollector(() -> Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+----
+
+The recommended way is to use an 
+link:user_guide.html#_advanced_externalizers[AdvancedExternalizer]
+as this provides the smallest payload.
+
+==== Parallel Computation
+
+Distributed streams by default try to parallelize as much as possible.  It is possible for the end user to control this and
+actually they always have to control one of the options.  There are 2 ways these streams are parallized.
+
+===== Local to each node
+When a stream is created from the cache collection the end user can choose between invoking
+link:https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#stream--[stream] or
+link:https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#parallelStream--[parallelStream]
+method.  Depending on if the parallel stream was picked will enable multiple threading for
+each node locally.  Note that some operations like a rehash aware iterator and forEach operations
+a sequential stream is forced locally.  The latter could be enhanced at some point to allow this.
```

Comment:
Making it more clear, basically it is ignored whether you have a parallel or sequential stream and a sequential stream is always used.